client/web/compose/src/components/PageBlocks/RecordListBase.vue

```diff
@@ -626,24 +626,40 @@
         v-if="showFooter"
         class="d-flex align-items-center justify-content-between p-2"
       >
-        <div class="text-truncate">
+        <div class="d-flex gap-3 align-items-center">
           <div
             v-if="options.showTotalCount"
-            class="ml-2 text-nowrap my-1"
+            class="ml-2 text-nowrap my-1 text-truncate"
           >
             <span
               v-if="pagination.count > options.perPage"
               data-test-id="pagination-range"
             >
               {{ $t('recordList.pagination.showing', getPagination) }}
             </span>
+
             <span
               v-else
               data-test-id="pagination-single-number"
             >
               {{ $t('recordList.pagination.single', getPagination) }}
             </span>
           </div>
+
+          <div
+            v-if="showPerPageSelector"
+            class="d-flex align-items-center gap-1 text-nowrap"
+          >
+            <span>
+              {{ $t('recordList.pagination.recordsPerPage') }}
+            </span>
+
+            <b-form-select
```

Comment:
The select will need a permanent width, otherwise its value can be hidden if its responsive